File: demo-service-spring-impl/src/main/java/com/github/zubmike/service/demo/types/PlanetarySystem.java
package com.github.zubmike.service.demo.types;

import org.hibernate.annotations.Cache;
import org.hibernate.annotations.CacheConcurrencyStrategy;
import org.hibernate.annotations.Immutable;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;
import java.io.Serial;

@Entity
@Table(name = "planetary_systems")
@Immutable
@Cache(region = "planetarySystem", usage = CacheConcurrencyStrategy.READ_ONLY)
public class PlanetarySystem extends BasicEntityDictItem {

	@Serial
	private static final long serialVersionUID = 7334333138985271731L;

	@Column(name = "code")
	private String code;

	public PlanetarySystem() {
	}

	public PlanetarySystem(int id, String name, String code) {
		super(id, name);
		this.code = code;
	}

	public String getCode() {
		return code;
	}

	public void setCode(String code) {
		this.code = code;
	}

}

File: demo-service-kotlin-impl/src/main/kotlin/com/github/zubmike/service/demo/types/PlanetarySystem.kt
package com.github.zubmike.service.demo.types

import org.hibernate.annotations.Cache
import org.hibernate.annotations.CacheConcurrencyStrategy
import org.hibernate.annotations.Immutable
import com.github.zubmike.service.types.BasicEntityDictItem

import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Table

@Entity
@Table(name = "planetary_systems")
@Immutable
@Cache(region = "planetarySystem", usage = CacheConcurrencyStrategy.READ_ONLY)
class PlanetarySystem : BasicEntityDictItem {

    @Column(name = "code")
    var code: String = ""

    constructor() {}

    constructor(id: Int, name: String?, code: String) : super(id, name) {
        this.code = code
    }

}
